src/ast-utils/ast-value.ts

```diff
@@ -106,9 +108,12 @@ function objectExpressionToValue(obj) {
 /**
  * BinaryExpressions are of the form "literal" + "literal"
  */
-function binaryExpressionToValue(member) {
+function binaryExpressionToValue(member: ESTree.BinaryExpression):(number|string) {
   if (member.operator == "+") {
-    return expressionToValue(member.left) + expressionToValue(member.right);
+    // We need to cast to `any` here because, while it's usually not the right
+    // thing to do to use '+' on two values of a mix of types because it's
+    // unpredictable, that is what the original code we're evaluating does.
+    return <any>expressionToValue(member.left) + expressionToValue(member.right);
```

Comment:
`expressionToValue` returns `string | number | boolean | RegExp`, and generally speaking you shouldn't `+` such things together, but we actually do want to do that in this case. The easiest way is to just cast the left value to `any` which appears to just make the local type inference give up completely.